intro-to-tests/practice-intro-to-pytest.md

```diff
@@ -0,0 +1,890 @@
+
+```python
+def test_returns_true_if_odd():
+    number = 5
+    result = is_odd(number)
+    assert result
+```
+
+##### !end-question
+##### !tests
+
+```py
+import unittest
+from main import *
+
+class TestPython1(unittest.TestCase):
+    def test_returns_true_if_odd(self):
+        number = 5
+        result = is_odd(number)
+        self.assertTrue(result)
+```
+
+##### !end-tests
+##### !hint
+
+Logic to find that something is odd or even is best solved using the modulo operator.
+
+##### !end-hint
+##### !hint
+
+Don't forget to put your code in a function with the right name and parameters.
+
+##### !end-hint
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 15 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: 71615b90-7562-480f-b868-50e23812a0f3
+* title: Intro to pytest
+##### !question
+
+Write a function to make the test pass. Use the information from the test to help determine what needs to be in the function body.
+
+```python
+def test_returns_false_if_even():
+    number = 6
+    result = is_odd(number)
+    assert not result
+```
+
+##### !end-question
+
+##### !tests
+
+```py
+import unittest
+from main import *
+
+class TestPython1(unittest.TestCase):
+    def test_returns_false_if_even(self):
+        number = 6
+        result = is_odd(number)
+        self.assertFalse(result)
+```
+
+##### !end-tests
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 16 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: 386b0492-77de-4132-a81c-8746090296c2
+* title: Intro to pytest
+##### !question
+
+Write a function to make the test pass. Use the information from the test to help determine what needs to be in the function body.
+
+```python
+def test_returns_None_if_negative():
+    number = -1000
+    result = is_odd(number)
+    assert result is None
+```
+
+##### !end-question
+##### !tests
+
+```py
+import unittest
+from main import *
+
+class TestPython1(unittest.TestCase):
+    def test_returns_None_if_negative(self):
+        number = -1000
+        result = is_odd(number)
+        assert result is None
+```
+
+##### !end-tests
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 17 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: a18259ec-848d-4ec4-99c1-c6a71ec1fab7
+* title: Intro to pytest
+
+##### !question
+
+Write a function to make the test pass. Use the information from the test to help determine what needs to be in the function body.
+
+```python
+def test_raises_runtime_error():
+    with pytest.raises(RuntimeError):
+        runs_mystery_algorithm()
+```
+
+##### !end-question
+##### !tests
+
+```py
+import unittest
+from main import *
+
+class TestPython1(unittest.TestCase):
+    def test_raises_runtime_error(self):
+        with self.assertRaises(RuntimeError):
+            p.runs_mystery_algorithm()
```

Comment:
What's this `p.`?  Don't know whether that's a `unittest` thing, a Learn thing, or a typo.
Also, make function name match if previous suggestion is taken.
```suggestion
            run_mystery_algorithm()
```

Reply:
It's a default Learn thing that I've been pulling out, and missed conforming this test. Thanks for the catch!